drivers/scheduler/k8s/k8s.go

```diff
@@ -4076,6 +4294,225 @@ func (k *K8s) GetTokenFromConfigMap(configMapName string) (string, error) {
 	return token, err
 }
 
+func (k *K8s) createCustomResourceObjects(
+	spec interface{},
+	ns *corev1.Namespace,
+	app *spec.AppSpec,
+) (interface{}, error) {
+
+	if obj, ok := spec.(*CustomResourceObjectYAML); ok {
+		log.Warn("applying custom resources")
+		cryaml := obj.Path
+		if _, err := os.Stat(cryaml); baseErrors.Is(err, os.ErrNotExist) {
+			return nil, fmt.Errorf("Cannot find yaml in path %s", cryaml)
+		}
+		cmdArgs := []string{"apply", "-f", cryaml, "-n", ns.Name}
+		err := osutils.Kubectl(cmdArgs)
+		if err != nil {
+			return nil, fmt.Errorf("Error applying spec [%s], Error: %s", cryaml, err)
+		}
+		obj.Namespace = ns.Name
+		obj.Name = "placeholder" //TODO1
+		return obj, nil
+	}
+
+	return nil, nil
+}
+
+func (k *K8s) destroyCustomResourceObjects(spec interface{}, app *spec.AppSpec) error {
+
+	if obj, ok := spec.(*CustomResourceObjectYAML); ok {
+		cryaml := obj.Path
+		if _, err := os.Stat(cryaml); baseErrors.Is(err, os.ErrNotExist) {
+			return &scheduler.ErrFailedToDestroyApp{
+				App:   app,
+				Cause: fmt.Sprintf("Failed to destroy Custom Resource Object: %v. Err: Cannot find yaml in path: %v", obj.Name, cryaml),
+			}
+		}
+
+		cmdArgs := []string{"delete", "-f", cryaml, "-n", obj.Namespace}
+		err := osutils.Kubectl(cmdArgs)
+		if err != nil {
+			return &scheduler.ErrFailedToDestroyApp{
+				App:   app,
+				Cause: fmt.Sprintf("Failed to destroy Custom Resource Object: %v. Err: %v", obj.Name, err),
+			}
+		} else {
+			log.Infof("[%v] Destroyed CustomResourceObject: %v", app.Key, obj.Name)
+			return nil
+		}
+	}
+
+	return nil
+}
+
+func (k *K8s) createApiExtensionsObjects(
```

Comment:
Shouldn't this be createCRDObjects?

Reply:
It could be, but my observation was that in **most** of the other _create_ functions, the name of the function corresponds to the "group" of the resources that it is creating. The group to which CRDs belong is `apiExtensions`, which is why I used that function. The convention folllowed by most other functions (like `createCoreObject`) is that, that function is used to create all types of objects in that _group_. 

But, having said that, I've just realized that `apiExtensions` (aka `apiextensions.k8s.io` group) has only one type of object (which is CRDs), so I _could_ rename the function. 
**Do you suggest that I do that?**

Reply:
Yes please rename it, so that it is evident

Reply:
✅